apps/bytebank-native/services/dasboard-service.ts

```diff
@@ -0,0 +1,222 @@
+        startOfThisWeek.setHours(0, 0, 0, 0);
+
+        const startOfLastWeek = new Date(startOfThisWeek);
+        startOfLastWeek.setDate(startOfLastWeek.getDate() - 7);
+
+        const endOfLastWeek = new Date(startOfThisWeek);
+
+        return date >= startOfLastWeek && date < endOfLastWeek;
+      }
+    }
+  }
+
+  private getIncomeByRange(transactions: ITransaction[], period: FilterType) {
+    const MONTHS = ["Jan", "Fev", "Mar", "Abr", "Mai", "Jun", "Jul", "Ago", "Set", "Out", "Nov", "Dez"];
+    const WEEK_DAYS = ["Dom", "Seg", "Ter", "Qua", "Qui", "Sex", "Sáb"];
+
+    switch (period) {
+      case FilterType.week: {
+        const { incomeByPeriod } = this.getValuesByRange(transactions, 7, (date) => date.getDay());
+        return WEEK_DAYS.map((day, i) => ({ period: day, income: incomeByPeriod[i] ?? 0 }));
+      }
+      case FilterType.month: {
+        const { incomeByPeriod } = this.getValuesByDay(transactions);
+        return incomeByPeriod.map((inc, i) => ({
+          period: String(i + 1).padStart(2, "0"),
+          income: inc ?? 0,
+        }));
+      }
+      case FilterType.year:
+      default: {
+        const { incomeByPeriod } = this.getValuesByRange(
+          transactions.filter((t) => new Date(t.date).getFullYear() === new Date().getFullYear()),
+          12,
+          (date) => date.getMonth()
+        );
+        return MONTHS.map((m, i) => ({ period: m, income: incomeByPeriod[i] ?? 0 }));
+      }
+    }
+  }
+
+  private getAmountAndExpensesByRange(transactions: ITransaction[], period: FilterType) {
+    const MONTHS = ["Jan", "Fev", "Mar", "Abr", "Mai", "Jun", "Jul", "Ago", "Set", "Out", "Nov", "Dez"];
+    const WEEK_DAYS = ["Dom", "Seg", "Ter", "Qua", "Qui", "Sex", "Sáb"];
+
+    switch (period) {
+      case FilterType.week: {
+        const { amountByPeriod, expensesByPeriod } = this.getValuesByRange(transactions, 7, (date) => date.getDay());
+        return WEEK_DAYS.map((day, i) => ({
+          period: day,
+          amount: amountByPeriod[i] ?? 0,
+          expenses: expensesByPeriod[i] ?? 0,
+        }));
+      }
+      case FilterType.month: {
+        const { amountByPeriod, expensesByPeriod } = this.getValuesByDay(transactions);
+        return amountByPeriod.map((amt, i) => ({
+          period: String(i + 1).padStart(2, "0"),
+          amount: amt ?? 0,
+          expenses: expensesByPeriod[i] ?? 0,
+        }));
+      }
+      case FilterType.year:
+      default: {
+        const { amountByPeriod, expensesByPeriod } = this.getValuesByRange(
+          transactions.filter((t) => new Date(t.date).getFullYear() === new Date().getFullYear()),
+          12,
+          (date) => date.getMonth()
+        );
+        return MONTHS.map((m, i) => ({
+          period: m,
+          amount: amountByPeriod[i] ?? 0,
+          expenses: expensesByPeriod[i] ?? 0,
+        }));
+      }
+    }
+  }
+
+  private getValuesByDay(transactions: ITransaction[]) {
```

Comment:
Fica mais legível dessa forma:

```suggestion
private getValuesByDay(transactions: ITransaction[]) {
  const now = new Date();
  const year = now.getFullYear();
  const month = now.getMonth();

  const daysInMonth = new Date(year, month + 1, 0).getDate();

  const isCurrentMonth = (date: Date) =>
    date.getFullYear() === year && date.getMonth() === month;

  const filtered = transactions.filter((t) => isCurrentMonth(new Date(t.date)));

  return this.getValuesByRange(
    filtered,
    daysInMonth,
    (date) => date.getDate() - 1
  );
}

```